prime-router/src/main/kotlin/azure/TokenFunction.kt

```diff
@@ -54,18 +54,29 @@ class TokenFunction(val metadata: Metadata = Metadata.getInstance()) : Logging {
         val tokenAuthentication = TokenAuthentication(DatabaseJtiCache(workflowEngine.db))
         if (tokenAuthentication.checkSenderToken(clientAssertion, senderKeyFinder, actionHistory)) {
             val token = tokenAuthentication.createAccessToken(scope, FindReportStreamSecretInVault(), actionHistory)
```

Comment:
kotlin's ability to get values out of if statements could be used here to simplify things:
```java
val response = if (tokenAuthentication.checkSenderToken(clientAssertion, senderKeyFinder, actionHistory)) {
...
} else {
...
}
actionHistory.trackActionResponse(response, null, workflowEngine.settings)
workflowEngine.recordAction(actionHistory)
return response

Reply:
Doing this worked and it's much nicer now!